pkg/leader/leader.go

```diff
@@ -109,7 +109,31 @@ func Become(ctx context.Context, lockName string) error {
 			log.Info("Became the leader.")
 			return nil
 		case apierrors.IsAlreadyExists(err):
-			log.Info("Not the leader. Waiting.")
+			existingOwners := existing.GetOwnerReferences()
+			switch {
+			case len(existingOwners) != 1:
+				log.Info("Leader lock configmap must have exactly one owner reference.", "ConfigMap", existing)
+			case existingOwners[0].Kind != "Pod":
+				log.Info("Leader lock configmap owner reference must be a pod.", "OwnerReference", existingOwners[0])
+			default:
+				leaderPod := &corev1.Pod{}
+				key = crclient.ObjectKey{Namespace: ns, Name: existingOwners[0].Name}
+				err = client.Get(ctx, key, leaderPod)
+				switch {
+				case apierrors.IsNotFound(err):
+					log.Info("Leader pod has been deleted, waiting for garbage collection do remove the lock.")
+				case err != nil:
+					return err
+				case isPodEvicted(*leaderPod):
```

Comment:
Before checking for `isPodEvicted()`, should we add another case for `isPodDeleted()`?

If there are lots of non-leader pods and deletion takes awhile, we would drastically reduce the number of unnecessary `Delete()` calls.  You can tell that the pod is deleted if `leaderPod.GetDeletionTimestamp() != nil`